process/smartContract/scrCommon/common.go

```diff
@@ -0,0 +1,75 @@
+package scrCommon
+
+import (
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/common"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/state"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+)
+
+// TestSmartContractProcessor is a SmartContractProcessor used in integration tests
+type TestSmartContractProcessor interface {
```

Comment:
hm, do we really need to have "testing" stuff in production code? if the component is really needed, maybe rename/refactor

Reply:
will be refactored in an upcoming PR

Reply:
see this comment
https://github.com/ElrondNetwork/elrond-go/pull/4496#discussion_r984480269